File: src/main/java/com/pulsesocial/service/CommentService.java
package com.pulsesocial.service;

import java.util.List;
import com.pulsesocial.dto.CommentRequest;
import com.pulsesocial.dto.CommentResponse;

public interface CommentService {
    String addComment(Long userId, Long postId, CommentRequest request);
    List<CommentResponse> getComments(Long postId);
    String deleteComment(Long commentId);
}

File: PulseSocial-Frontend/src/services/commentService.js
import api from "../api/axios";

export const addComment = (postId, content, userId) => {
    const url = userId ? `/comments?postId=${postId}&userId=${userId}` : `/comments?postId=${postId}`;
    return api.post(url, { content });
};

export const getComments = (postId) => {
    return api.get(`/comments/post/${postId}`);
};

export const deleteComment = (commentId) => {
    return api.delete(`/comments/${commentId}`);
};
